x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/engines_list_logic.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { HttpError } from '../../../../../common/types/api';
+
+import { flashAPIErrors } from '../../../shared/flash_messages';
+
+import { FetchEnginesAPILogic } from '../../api/engines/fetch_engines_api_logic';
+
+import { DEFAULT_META, EngineListDetails, Meta, updateMetaPageIndex } from './types';
+
+export interface EnginesListActions {
+  apiError(error: HttpError): HttpError;
+  apiSuccess({ enginesList, meta }: { enginesList: EngineListDetails[]; meta: Meta }): {
```

Comment:
suggestion: Rather than manually defining these API types, ideally we import them from the API logic itself so that they don't accidentally drift. You do this with special types, like we do here: https://github.com/elastic/kibana/blob/1d0a8c9ddd47fc6b9a70e3b46bb15ef0553ce18f/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/connector/connector_configuration_logic.ts#L33